### ISSUE TITLE
Email: WCAG 3.0 public comment (1 - question on backward compatibility)

### ISSUE THREAD

**Issue by jspellman** [w3c/silver#424]
[Comment from Email](https://lists.w3.org/Archives/Public/public-agwg-comments/2021Feb/0010.html):
from [spreadsheet](https://lists.w3.org/Archives/Public/public-agwg-comments/2021Feb/att-0010/20210226drempelvrij_Foundation_-_comments_on_working_drafts_WCAG_3.0.xlsx) 

WCAG 3.0   does not want to advance the WCAG 2.0 requirement: "Ensure that the   revision is 'backwards and forward compatible '"   . The intention is to include WCAG 2.x content, but migrate it to a different   structure and conformance model.

Does   this mean that websites that comply with WCAG 2.x have to start from scratch   in order to meet with WCAG 3.0 guidelines?

**Comment by jspellman** [comment]
Thank you for your comment. Project members are working on your comment. You may see discussion in the comment thread and we may ask for additional information as we work on it. We will mark the official response when we are finished and close the issue.

**Comment by alastc** [comment]
Proposed response:

<hr>

The backwards compatibility requirement would mean that the testing & conformance approach between versions are compatible, pass a new version would mean passing a previous version.

Websites would not have to implement accessibility in a completely different way, the guidance from WCAG 2.x will be translated to fit the WCAG 3 structure and approach.

However, the measurement and testing approach may be very different, which is why WCAG 3 is not taking on the backwards compatibility requirement (with the previous guidelines & conformance) as a requirement.

**Comment by Myndex** [comment]
Hi Jeanne, Alastair @jspellman @alastc 

I'd like to add just a brief amount of perspective for clarity, and to mention some important implications or barriers. If this is not the appropriate thread please feel free to move  this post.

## Some Definitions
- **Backwards compatible:** a site that passes WCAG 3.0 at some pre-determined level, will automatically pass all tests of WCAG 2.0 (or 2.x?) at the AA level.

- **Forwards compatible:** a site that passes WCAG 2 at the AA level would automatically pass all tests at least at the lowest level of conformance of WCAG 3.0.

- **Forward grandfathering:** pages with a "grandfathered" meta tag are tested as 2.0, AND/OR exception(s) are added to WCAG 3.0 to allow a WCAG 2.x AA site to pass 3.0 with those exception(s) at the lowest level of conformance.

- **Sectional transitioning:** a site transitions from WCAG 2.x to WCAG 3.0 on a **_per SC_** or per section basis.

-------
## DISCUSSION
### _BACKWARDS_ 
- Backwards compatible is a non-starter as it would require portions of WCAG 2.x to be re-written or specific exceptions made to accommodate the newer technologies of WCAG 3.0. As I understand it, even a WCAG 2.**3** can not change any existing SCs or override them.
    - Case in point, WCAG 2 contrast has some issues calculating some contrasts. False passes are not an issue as APCA would correctly fail those. But certain bright colors will pass with APCA as they should pass, nevertheless due to the math, WCAG 2 would indicate a fail for these.
    - Myself **I object to a backwards compatible WCAG 3** if WCAG 2 can not be modified, I think it would be terrible to hobble WCAG 3 guidelines into those SCs.
    - On the other hand, if there was a WCAG 2.3 that included backwards compatible "overrides" such as to allow WCAG 3.5 to override 1.4.3 and 1.4.11 etc., this might be more palatable...
    -ut what utility is there for taking a site that passes 3.0 and testing it for 2.1 ?? 

### _FORWARDS_ 
- Forwards compatible is doable with the caveat that the WCAG 3.0 guidelines would have to have a conformance level that was structured to pass WCAG 2. We developed that kind of conformance for visual contrast, making "score 1" a WCAG 2 catch all, and in most cases would be effective, but if someone had designed a passing site on the edge of conformance, they could still fail the score 1 level.
    - Right now I'd call this the _"well designed sites that pass WCAG 2 AA would pass 3.0 at least at score 1"_ model.
    - The drawback is making conformance potentially weak in some areas, or causing other conflicts.

### _GRANDFATHERING_ 
- Grandfathering forward, presumably with some restrictions, is the most practical and simplest IMO, while doing the best to maintain integrity of WCAG 3.0. In this case no guidelines are "weakened" just to allow a pass.
    - But instead, a page with a... **`  <meta wcag3="2.0-grandfathered">   `** ....2.0-grandfathered tag gets tested under 2.0, otherwise gets tested under 3.0. Presumably some SCs might be excluded from grandfathering.
    - Alternately or inclusively each guideline might have specific items that are ecluded as grandfathered.
    - grandfathering should probably have an end condition, such as a page or asset is substantially changed in form, or by a certain future date, or declared "archival" and frozen in state.  

### _SECTIONAL_
- **Sectional transitioning:** a site transitions from WCAG 2.x to WCAG 3.0 on a **_per SC_** or per section basis.
    - Also known as _Andy's much too complicated version._ ... I only mention as this seems like a need or desired path for very large sites. 
   - For example purposes only, let's say the site first wants to convert color and contrast, so they replace the 1.4.3 series of SCs with the 3.5 Visual Contrast. During this time, they could test for conformance using WCAG 3.0 Visual Contrast instead of the 1.4.3 series, but still use the WCAG 2.x SCs for other conformance categories that have not transitioned yet.
    - To support this workflow, some meta-tags are needed to indicate which spec a page should be tested under.


Thank you for reading,

Andy

**Comment by alastc** [comment]
The group approved the response above. @Myndex we've opened a new issue on backwards compatibility for after WCAG 3 is published, #569
